### ISSUE TITLE
Review and address accessibility issues

### ISSUE THREAD

**Issue by jaymcp** [bigbite/image-comparison#4]
The block requires an accessibility review in both the editor and the frontend.

Known issues at time of writing:
- the block container has no descriptive labelling on the frontend
- the caption should be always-enabled, to encourage users to add a descriptive explanation of their visualisations, so no setting is required
- the divider button needs labelling (aria label / screen reader text contents) or marking as presentational in both the editor and frontend

Testing should be carried out with keyboard, screen reader, and automated tools like wave or axe
